sdk/keyvault/keyvault-keys/src/cryptographyClient.ts

```diff
@@ -0,0 +1,524 @@
+import {
+  NewPipelineOptions,
+  isNewPipelineOptions,
+  Pipeline,
+} from "./core/keyVaultBase";
+import { KeyVaultClient } from "./core/keyVaultClient";
+import { challengeBasedAuthenticationPolicy } from "./core/challengeBasedAuthenticationPolicy";
+import * as crypto from "crypto";
+import * as constants from "constants";
+
+let keyto: any;
+if (isNode) {
+  keyto = require("@trust/keyto");
+}
+
+/**
+ * The client to interact with the KeyVault cryptography functionality
+ */
+export class CryptographyClient {
+
+  /**
+   * Retrieves the complete key from the key vault
+   * @param options Options for retrieving key
+   */
+  public async getKey(options?: GetKeyOptions): Promise<JsonWebKey> {
+    if (typeof this.key === "string") {
+      if (!this.name || this.name === "") {
+        throw new Error("getKey requires a key with a name");
+      }
+      const key = await this.client.getKey(
+        this.vaultBaseUrl,
+        this.name,
+        options && options.version ? options.version : (this.version ? this.version : ""),
+        options
+      );
+      return key.key!;
+    } else {
+      return this.key;
+    }
+  }
+
+  /**
+   * Encrypts the given plaintext with the specified cryptography algorithm
+   * @param plaintext The text to encrypt
+   * @param algorithm The algorithm to use
+   * @param options Additional options
+   */
+  public async encrypt(
+    plaintext: Uint8Array,
+    algorithm: JsonWebKeyEncryptionAlgorithm,
+    options?: EncryptOptions
+  ): Promise<Uint8Array> {
+    if (isNode) {
+      await this.fetchFullKeyIfPossible();
+
+      if (typeof this.key !== "string") {
+        switch (algorithm) {
+          case "RSA1_5": {
+            let keyPEM = keyto.from(this.key, "jwk").toString('pem', 'public_pkcs1');
+
+            let padded: any = { key: keyPEM, padding: constants.RSA_PKCS1_PADDING };
+            const encrypted = crypto.publicEncrypt(padded, Buffer.from(plaintext));
+            return encrypted;
+          };
+          case "RSA-OAEP": {
+            let keyPEM = keyto.from(this.key, "jwk").toString('pem', 'public_pkcs1');
+
+            const encrypted = crypto.publicEncrypt(keyPEM, Buffer.from(plaintext));
+            return encrypted;
+          };
+        }
+      }
+    }
+
+    // Default to the service
+    let result = await this.client.encrypt(this.vaultBaseUrl, this.name, this.version, algorithm, plaintext, options);
+    return result.result!;
+  }
+
+  /**
+   * Decrypts the given ciphertext with the specified cryptography algorithm
+   * @param ciphertext The ciphertext to decrypt
+   * @param algorithm The algorithm to use
+   * @param options Additional options
+   */  
+  public async decrypt(
+    ciphertext: Uint8Array,
+    algorithm: JsonWebKeyEncryptionAlgorithm,
+    options?: DecryptOptions
+  ): Promise<Uint8Array> {
+    let result = await this.client.decrypt(this.vaultBaseUrl, this.name, this.version, algorithm, ciphertext, options);
```

Comment:
for now it's ok to just defer decrypt to the service, but eventually we could check if the key material is sufficient to preform the operation.  Perhaps we should create an issue to track this.

Reply:
Added https://github.com/Azure/azure-sdk-for-js/issues/4566